core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/ClusterUpgradeImageFilter.java

```diff
@@ -85,23 +83,27 @@ private boolean isOsVersionsMatch(Image currentImage, Image newImage) {
     }
 
     private Predicate<Image> validateCmVersion(Image currentImage) {
-        return image -> compareVersionStrict(currentImage.getPackageVersions().get(CM_PACKAGE_KEY), image.getPackageVersions().get(CM_PACKAGE_KEY));
+        return image -> upgradePermissionProvider.permitCmAndSatckUpgrade(currentImage.getPackageVersions().get(CM_PACKAGE_KEY),
+                image.getPackageVersions().get(CM_PACKAGE_KEY));
     }
 
     private Predicate<Image> validateStackVersion(Image currentImage) {
-        return image -> compareVersionStrict(currentImage.getPackageVersions().get(STACK_PACKAGE_KEY), image.getPackageVersions().get(STACK_PACKAGE_KEY));
+        return image -> upgradePermissionProvider.permitCmAndSatckUpgrade(currentImage.getPackageVersions().get(STACK_PACKAGE_KEY),
```

Comment:
Can you please extract the upgradePermissionProvider.permitCmAndSatckUpgrade call into a new method?

Reply:
Thanks. Done.